.github/workflows/coding-standards.yml

```diff
@@ -0,0 +1,68 @@
+name: "Coding Standards"
+
+on:
+  pull_request:
+    branches:
+      - "v*.*"
+      - "master"
+  push:
+    branches:
+      - "v*.*"
+      - "master"
+
+jobs:
+  coding-standards:
+    name: "Coding Standards"
+    runs-on: "ubuntu-20.04"
+
+    strategy:
+      matrix:
+        php-version:
+          - "7.4"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+
+      - name: Setup cache environment
+        id: extcache
+        uses: shivammathur/cache-extensions@v1
```

Comment:
Are there any concerns with depending on third-party repositories for action recipes? I imagine if this was taken down our CI builds would break (among many others). Might there be security concerns we were using encrypted environment variables (e.g. Atlas testing) and an action dependency was compromised?

AFAIK, secrets are only tested with Evergreen so this may be a hypothetical question.

Reply:
> Are there any concerns with depending on third-party repositories for action recipes? I imagine if this was taken down our CI builds would break (among many others).

Yes, our CI builds would break in that case, but I'd expect a fork to spring up relatively soon considering how widely used these actions are.

> Might there be security concerns we were using encrypted environment variables (e.g. Atlas testing) and an action dependency was compromised?

Good question, I haven't considered yet. Quoting from the [secrets documentation](https://docs.github.com/en/actions/reference/encrypted-secrets):
> To make a secret available to an action, you must set the secret as an input or environment variable in the workflow file. [...] You can use and read encrypted secrets in a workflow file if you have access to edit the file.

Reading this, any secrets defined in our org or repo would have to be passed on to third-party actions as input or environment variables (either using `with` or `env` keys). I'd assume that a compromised third-party action file would not gain access to secrets defined in our repository, but I concede I haven't tested this (yet).

That said, I haven't configured any secrets yet, so we're not running any AWS tests or anything. We can decide to do this at a later date, or we can leave testing with secrets to our evergreen pipeline.

Reply:
> That said, I haven't configured any secrets yet, so we're not running any AWS tests or anything. We can decide to do this at a later date, or we can leave testing with secrets to our evergreen pipeline.

I'd say nothing to do here, but something to keep in mind if you have grander plans for moving drivers onto GitHub Actions down the line. Maybe you want to add a line about this in your scope doc, but I reckon it's potentially future work since you were only focused on a subset of tests for quick PR feedback.

Reply:
Will do to make sure other engineers are aware, although I want this to be a subset for quick PR feedback in all drivers for the time being.